### DISCUSSION TITLE
Is a valid schema URL still required when building from a string?

### DISCUSSION THREAD

**Discussion by markgomez** [santhosh-tekuri/boon#3] [Q&A] [answered]
It's great to see another JSON Schema validator in Rust! Some quick questions: in the ["from strings" example](https://github.com/santhosh-tekuri/boon/blob/d466730e5e5c7c663bd6739e74e39d1e2f7baae4/tests/examples.rs#L22-L23) I noticed a `schema_url` is specified, but if the schema is being built from a string literal (i.e., not from a local or remotely hosted `.json` file), what is its purpose here? Is it just some kind of "placeholder?" Will `http://tmp/schema.json` (or any other placeholder URL) be ignored in this situation?

**Comment by santhosh-tekuri** [comment, marked as answer]
`schema_url` must be specified. From the example it might look like, it has no purpose.

consider you have two schema files `./animals/cat.json`, `./animals/dog.json` in your current working directory. and you want to check if the json matches one of these two schemas. then use the following schema url and literal:

```rust
let schema_url = "animals/temp.json"
let schema = json!({
    "oneOf": [
         { "$ref": "cat.json" },
         { "$ref": "dog.json"}
    ]
});
```

let us say you want to check with `http://animals.com/cat.json`, `http://animals.com/dog.json` then, just change the `schema_url` to `http://animals.com/temp.json`

many libraries don't ask for `schema_url` for schema literals. but it is required for resolving references within schema.

as per specification each schema must be identified by canonical url. the canonical url is used in resolving references and also in error reporting.

I guess the current example should be updated to show significance of `schema_url`
